hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterStateManager.java

```diff
@@ -104,26 +104,28 @@ void initialClusterState(ClusterState initialState) {
                         + initialState);
                 return;
             }
-            this.state = initialState;
-            changeNodeState(initialState);
-            node.getNodeExtension().onClusterStateChange(initialState, false);
+            doSetClusterState(initialState, true);
```

Comment:
Beware that here the method call will additionally call `stateLockRef.set(ClusterStateLock.NOT_LOCKED);` in contrast to the old code.